admission/hooks/validate_application.go

```diff
@@ -113,8 +113,16 @@ func (v *argocdApplicationValidator) extractRepoURLs(app *unstructured.Unstructu
 		}
 	}
 
+	hydratorRepoURL, found, err := unstructured.NestedString(app.UnstructuredContent(), "spec", "sourceHydrator", "drySource", "repoURL")
+	if err != nil {
+		return nil, fmt.Errorf("unable to get spec.sourceHydrator.drySource.repoURL; %w", err)
+	}
+	if found {
+		repoURLs = append(repoURLs, hydratorRepoURL)
+	}
+
 	if len(repoURLs) == 0 {
-		return nil, errors.New("spec.source.repoURL nor spec.sources not found")
+		return nil, errors.New("spec.source.repoURL, spec.sources, or spec.sourceHydrator.drySource.repoURL not found")
```

Comment:
The error condition is not appropriate.
It should raise an error when both `spec.source.repoURL` and `spec.sources` are empty, but currently, it does not raise an error when both `spec.source.repoURL` and `spec.sources` are empty while `spec.sourceHydrator.drySource.repoURL` is not, which is incorrect.